include/sensors/core/mmr920C04.hpp

```diff
@@ -0,0 +1,281 @@
+        auto message = can_messages::ReadFromSensorResponse{
+            .sensor = get_sensor_id(), .sensor_data = temperature};
+        can_client.send_can_message(get_host_id(), message);
+    }
+
+    auto send_status() -> void {
+        auto status = Status::to_status(_registers.status.reading);
+        auto message = can_messages::ReadFromSensorResponse{
+            .sensor = get_sensor_id(),
+            .sensor_data = static_cast<int32_t>(status)};
+        can_client.send_can_message(get_host_id(), message);
+    }
+
+    auto send_threshold() -> void {
+        auto message = can_messages::SensorThresholdResponse{
+            .sensor = get_sensor_id(), .threshold = get_threshold()};
+        can_client.send_can_message(get_host_id(), message);
+    }
+
+    // callbacks
+    static auto handle_data(const sensor_callbacks::MaxMessageBuffer &buffer,
+                            MMR92C04 *instance, Registers reg) -> void {
+        uint32_t data = 0x0;
+        const auto *iter = buffer.cbegin();
+        iter = bit_utils::bytes_to_int(iter, buffer.cend(), data);
+        switch (reg) {
+            case Registers::PRESSURE_READ:
+                instance->read_pressure(data);
+                break;
+            case Registers::LOW_PASS_PRESSURE_READ:
+                instance->read_pressure_low_pass(data);
+                break;
+            case Registers::TEMPERATURE_READ:
+                instance->read_temperature(data);
+                break;
+            case Registers::STATUS:
+                instance->read_status(data);
+                break;
+            case Registers::RESET:
+            case Registers::IDLE:
+            case Registers::MEASURE_MODE_1:
+            case Registers::MEASURE_MODE_2:
+            case Registers::MEASURE_MODE_3:
+            case Registers::MEASURE_MODE_4:
+            case Registers::MACRAM_WRITE:
+                break;
+        }
+    }
+
+    static auto send_to_can(MMR92C04 *instance, Registers reg) {
+        switch (reg) {
+            case Registers::PRESSURE_READ:
+                instance->send_pressure();
+                break;
+            case Registers::LOW_PASS_PRESSURE_READ:
+                instance->send_pressure_low_pass();
+                break;
+            case Registers::TEMPERATURE_READ:
+                instance->send_temperature();
+                break;
+            case Registers::STATUS:
+                instance->send_status();
+                break;
+            case Registers::RESET:
+            case Registers::IDLE:
+            case Registers::MEASURE_MODE_1:
+            case Registers::MEASURE_MODE_2:
+            case Registers::MEASURE_MODE_3:
+            case Registers::MEASURE_MODE_4:
+            case Registers::MACRAM_WRITE:
+                break;
+        }
+    }
+
+  private:
+    MMR920C04RegisterMap _registers{};
+    bool _initialized = false;
+    int32_t threshold = 0x8;
```

Comment:
Is this a constant? What is 8? What are the units?

Reply:
This is not a constant, it can be set from CAN. I'll put the units in a comment.

Reply:
Better to name the variable with the unit.